log4j-core/src/main/java/org/apache/logging/log4j/core/config/properties/PropertiesConfigurationBuilder.java

```diff
@@ -207,7 +221,12 @@ private ScriptFileComponentBuilder createScriptFile(final Properties properties)
     }
 
     private AppenderComponentBuilder createAppender(final String key, final Properties properties) {
-        final String name = (String) properties.remove(CONFIG_NAME);
+        final String name;
+        if (properties.containsKey(CONFIG_NAME)) {
+            name = (String) properties.remove(CONFIG_NAME);
+        } else {
+            name = key;
+        }
```

Comment:
I think we may be slightly better off retaining the original `remove` logic, only traversing the map once:
```suggestion
        final String name = Objects.toString(properties.remove(CONFIG_NAME), key);
```